guides/ci/travis.md

```diff
@@ -0,0 +1,252 @@
+
+### Using `latest`, `nightly` and a specific Crystal release all together!
+
+For using 3 versions of the compiler we are going to use a different configuration file based on [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#build-matrix)
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+## Installing shards packages
+
+Thanks to the `language: crystal` support, Travis CI automatically runs `shards install`. And for faster ci times we may use [caching](#caching)
+
+#### Using Docker
+
+When using a containerization service like Docker we will need to run `shards install` explicitly, like this:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 shards install
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+**Note:** Since the shards will be installed in `./lib/` folder, it will be preserved for the second docker run command.
+
+## Installing binary dependencies
+
+> It is important to mention that there are many ways to achieve this and **it will heavily depend on our development workflow**.
+
+Our application or maybe some shards may required libraries and packages. In that case we will need to install them using `apt` (or similar).
+
+Here is a first example installing SQLite3 using Travis CI configuration file:
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+before_install:
+  - sudo apt-get -y install libsqlite3-dev
+
+addons:
+  apt:
+    update: true
+
+script:
+  - crystal spec
+```
+
+#### Using Docker
+
+The same can be accomplished using a containerization service like Docker. Here is an example of a `Dockerfile` installing SQLite3:
+
+```dockerfile
+# Dockerfile
+FROM crystallang/crystal:latest
+
+RUN apt-get update && apt-get install -y libsqlite3-dev
+```
+
+And here is the Travis CI configuration file:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+before_install:
+  # build image using Dockerfile:
+  - docker build -t testing .
+
+script:
+  # run specs in the container
+  - docker run -v $PWD:/src -w /src testing crystal spec
+```
+
+**Note:** Dockerfile arguments can be used to use the same Dockerfile for latest, nightly or a specific version.
+
+## Using a database (example using MySQL)
+
+As you may notice, in the last configuration file we are using `docker` as a `service`. Travis CI may start different services as needed. For example, we may need [MySQL](https://docs.travis-ci.com/user/database-setup/#mysql).
```

Comment:
I'm not sure the reference to the previous section is helpful. Readers who are only interested in travis runners have probably skipped it anyways. Maybe we could just swap dependencies and services section? This would avoid applying a concept before its introduction. Services like databases are probably more often used.